src/core/ext/xds/xds_api.cc

```diff
@@ -971,9 +971,9 @@ absl::StatusOr<XdsApi::ResourceName> XdsApi::ParseResourceName(
 std::string XdsApi::ConstructFullResourceName(absl::string_view authority,
                                               absl::string_view resource_type,
                                               absl::string_view name) {
-  if (absl::StartsWith(name, "xdstp:")) {
-    return absl::StrCat("xdstp://", authority, "/", resource_type,
-                        absl::StripPrefix(name, "xdstp:"));
+  if (absl::StartsWith(authority, "xdstp:")) {
```

Comment:
This can be done a bit more efficiently if written as:

```
if (absl::ConsumePrefix(&authority, "xdstp:")) {
  return absl::StrCat("xdstp://", authority, "/", resource_type, "/", name);
}
```

Essentially, `absl::ConsumePrefix()` avoids scanning the string twice, once in `absl::StartsWith()` and again in `absl::StripPrefix()`.